docs/en/docs/encoders.md

```diff
@@ -0,0 +1,140 @@
+* `DateEncoder` - Serializes date and datetime objects.
+* `StructureEncoder` - Serializes more complex data types. `set, frozenset, GeneratorType, tuple, deque`.
+
+What a brand new encoder is needed and it is not natively supported by Lilya? Well, [building a custom encoder](#build-a-custom-encoder)
+is extremly easy and possible.
+
+## Build a custom encoder
+
+As mentioned before, Lilya has [default encoders](#default-encoders) that are used to transform a response
+into a `json` serialisable response.
+
+To build a custom encoder you must use the `Encoder` class from Lilya and override the `serialize()` function
+where it applies the serialisation process of the encoder type.
+
+Then you **must register the encoder** for Lilya to use it.
+
+When defining an encoder the `__type__` or `def is_type(self, value: Any) -> bool:`
+**must be declared or overridden**.
+
+When the `__type__` is properly declared, the default `is_type` will evaluate the object against the
+type and return `True` or `False`.
+
+This is used internally to understand the type of encoder that will be applied to a given object.
+
+!!! warning
+    If you are not able to provide the `__type__` for any reason and you just want to override the
+    default evaluation process, simple override the `is_type()` and apply your custom logic there.
+
+    E.g.: In Python 3.8, for a Pydantic `BaseModel` if passed in the `__type__`, it will throw an
+    error due to Pydantic internals, so to workaround this issue, you can simply override the `is_type()`
+    and apply the logic that validates the type of the object and returns a boolean.
+
+```python
+from lilya.encoders import Encoder, register_encoder
+```
+
+**Example**
+
+Create and register an encoder that handles `msgspec.Struct` types.
+
+```python
+{!> ../../../docs_src/responses/encoders/example.py !}
+```
+
+Simple right? Because now the `MsgSpecEncoder` is registered, you can simply do this in your handlers
+and return **directly** the `msgspec.Struct` object type.
+
+```python
+from msgspec import Struct
+
+from lilya.routing import Path
+
+
+class User(Struct):
+    name: str
+    email: str
+
+
+def msgspec_struct():
+    return User(name="lilya", url="example@lilya.dev")
+```
+
+### Design specific custom encoders
+
+**Lilya being 100% pure python and not tight to any particular validation library** allows you to
+design custom encoders that are later used by Lilya responses.
+
+Ok, this sounds a bit confusing right? I bet it does so let us go slowly.
+
+Imagine you want to use a particular validation library such as [Pydantic](https://pydantic.dev/),
+[msgspec](https://jcristharif.com/msgspec/) or even [attrs](https://www.attrs.org/en/stable/) or something
+else at your choice.
+
+You want to make sure that if you return a pydantic model or a msgspec Struct or even a `define` attr class.
+
+Let us see how it would look like for all of them.
+
+**For pydantic**
+
+Nothing required anymore. Works out of the box.
```

Comment:
We still want to show an example here. Why? Because its docs and we want people to understand. So we need to change to:

Prior to version 0.11.0, it would be require to design one but after version 0.11.0, the internal mechanism for the encoders was changed in a way that encompasses even the Pydantic Base models making it not "required" (although you can still build your own) anymore and working out of the box.

For example purposes, this is how it would look like building a specific Pydantic encoder:

ADD DOCS SRC PYDANTIC FILE with updated version here.

Reply:
done